src/components/shared/Icon.js

```diff
@@ -41,7 +41,7 @@ class Icon extends PureComponent<Props> {
     }
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
+  componentDidUpdate(nextProps: Props) {
```

Comment:
`componentDidUpdate` doesn't take the next props, but the previous props. As a result this change doesn't do what it's supposed to do, sadly, and we may find missing icons when scrolling the call tree.

Please look at the documentation of React so that you understand what you're doing:
https://reactjs.org/docs/react-component.html#componentdidupdate.

Because `componentDidUpdate` happens after a render, `this.props` already contains the new value, so I think we want to use `this.props` here.

Reply:
Thankyou I was about to make this change. As I saw more about it.
In `componentDidUpdate()`  this.props points to the newest props, so we are simply replacing the newProps with this.props